programs/bpf_loader/src/syscalls/mem_ops.rs

```diff
@@ -108,19 +108,20 @@ declare_builtin_function!(
                 n,
                 invoke_context.get_check_aligned(),
             )?;
-            let cmp_result = translate_type_mut::<i32>(
-                memory_mapping,
-                cmp_result_addr,
-                invoke_context.get_check_aligned(),
-            )?;
```

Comment:
In this case, would it be possible to make the mutable translation first?

I wonder if we could follow the same pattern everywhere and then write a comment instructing about the correct usage.